js/ui/calendar/ui.calendar.js

```diff
@@ -1178,9 +1267,18 @@ const Calendar = Editor.inherit({
         delete this._suppressNavigation;
     },
 
+    _processValueChanged: function(value, previousValue) {
+        const multiselect = this.option('multiselect');
+        value = multiselect ? value.map((item) => this._convertToDate(item)) : [this._convertToDate(value)];
```

Comment:
There are a lot of `if-else` and ternary operators because of `multiselect`. I think we should think about using of [Strategy design pattern](https://refactoring.guru/design-patterns/strategy)